test/plugins/FakeTimeStampDataGenerator.cpp

```diff
@@ -0,0 +1,190 @@
+/**
+ * @file FakeTimeStampDataGenerator.cpp FakeTimeStampDataGenerator class
+ * implementation
+ *
+ * This is part of the DUNE DAQ Software Suite, copyright 2020.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+
+#include "CommonIssues.hpp"
+#include "FakeTimeStampDataGenerator.hpp"
+
+#include "appfwk/app/Nljs.hpp"
+
+#include "logging/Logging.hpp"
+
+#include <chrono>
+#include <cstdlib>
+#include <thread>
+#include <string>
+#include <vector>
+
+using pd_clock = std::chrono::duration<double, std::ratio<1, 50000000>>;
+
+namespace dunedaq {
+namespace trigger {
+
+FakeTimeStampDataGenerator::FakeTimeStampDataGenerator(const std::string& name)
+  : dunedaq::appfwk::DAQModule(name)
+  , thread_(std::bind(&FakeTimeStampDataGenerator::do_work, this, std::placeholders::_1))
+  , outputQueue_()
+  , queueTimeout_(100)
+  , generator()
+{
+  register_command("conf",  &FakeTimeStampDataGenerator::do_configure);
+  register_command("start", &FakeTimeStampDataGenerator::do_start);
+  register_command("stop",  &FakeTimeStampDataGenerator::do_stop);
+  register_command("scrap", &FakeTimeStampDataGenerator::do_scrap);
+}
+
+void
+FakeTimeStampDataGenerator::init(const nlohmann::json& init_data)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering init() method";
+  auto ini = init_data.get<appfwk::app::ModInit>();
```

Comment:
I think this block is more clearly written using `appfwk::queue_inst` (from `appfwk/DAQModuleHelper.hpp`). See an example here:

https://github.com/DUNE-DAQ/trigger/blob/e2bdf8e3522a5df12a854bfa940fa08afa98e471/plugins/FakeDataFlow.cpp#L40

You'll need to pick a name for the queue.